modules/hostap/src/supp_api.c

```diff
@@ -239,6 +239,18 @@ int z_wpa_supplicant_connect(const struct device *dev,
 	_wpa_cli_cmd_v("set_network %d scan_ssid 1", resp.network_id);
 	_wpa_cli_cmd_v("set_network %d key_mgmt NONE", resp.network_id);
 	_wpa_cli_cmd_v("set_network %d ieee80211w 0", resp.network_id);
+	if ((params->band & BIT(WIFI_FREQ_BAND_2_4_GHZ)) == BIT(WIFI_FREQ_BAND_2_4_GHZ)) {
```

Comment:
This capability of setting channels, should be in a separate commit.

Reply:
And have you tried skipping this altogether and let WPA supplicant derive these from the driver's capability? Its not ideal to duplicate these esp. without the regulatory domain information.

Reply:
Should we then remove band configuration support from stored wifi credentials as well?

Reply:
Well, we can still keep it and implement, but we need to dynamically query the supported channels from the driver and use that rather than a static list.

Reply:
Done